core/src/banking_stage/consume_worker.rs

```diff
@@ -170,21 +170,31 @@ fn try_drain_iter<T>(work: T, receiver: &Receiver<T>) -> impl Iterator<Item = T>
 /// done.
 pub(crate) struct ConsumeWorkerMetrics {
     id: String,
-    interval: AtomicInterval,
     has_data: AtomicBool,
+    slot: AtomicU64,
 
     count_metrics: ConsumeWorkerCountMetrics,
     error_metrics: ConsumeWorkerTransactionErrorMetrics,
     timing_metrics: ConsumeWorkerTimingMetrics,
 }
 
 impl ConsumeWorkerMetrics {
-    /// Report and reset metrics iff the interval has elapsed and the worker did some work.
-    pub fn maybe_report_and_reset(&self) {
-        const REPORT_INTERVAL_MS: u64 = 1000;
-        if self.interval.should_update(REPORT_INTERVAL_MS)
-            && self.has_data.swap(false, Ordering::Relaxed)
-        {
+    /// Report and reset metrics when the worker did some work and:
+    /// a) (when a leader) Previous slot is not the same as current.
+    /// b) (when not a leader) report the metrics accumulated so far.
+    pub fn maybe_report_and_reset(&self, slot: Option<Slot>) {
+        if let Some(slot) = slot {
+            let prev_slot_id: u64 = self.slot.load(Ordering::Relaxed);
+            if slot != prev_slot_id {
+                if !self.has_data.swap(false, Ordering::Relaxed) {
+                    return;
+                }
+                self.count_metrics.report_and_reset(&self.id);
+                self.timing_metrics.report_and_reset(&self.id);
+                self.error_metrics.report_and_reset(&self.id);
+                self.slot.swap(slot, Ordering::Relaxed);
+            }
+        } else {
```

Comment:
this will spam, we need to check if there was or was not a previous slot stored in the metrics

Reply:
every time the scheduler loop runs it may call into this function. this function needs to check if the slot is different than what was previously there